modules/samtools/fixmate/main.nf

```diff
@@ -0,0 +1,44 @@
+// Import generic module functions
+include { initOptions; saveFiles; getSoftwareName; getProcessName } from './functions'
+
+params.options = [:]
+options        = initOptions(params.options)
+
+process SAMTOOLS_FIXMATE {
+    tag "$meta.id"
+    label 'process_low'
+    publishDir "${params.outdir}",
+        mode: params.publish_dir_mode,
+        saveAs: { filename -> saveFiles(filename:filename, options:params.options, publish_dir:getSoftwareName(task.process), meta:meta, publish_by_meta:['id']) }
+
+    conda (params.enable_conda ? "bioconda::samtools=1.14" : null)
+    if (workflow.containerEngine == 'singularity' && !params.singularity_pull_docker_container) {
+        container "https://depot.galaxyproject.org/singularity/samtools:1.14--hb421002_0"
+    } else {
+        container "quay.io/biocontainers/samtools:1.14--hb421002_0"
+    }
+
+    input:
+    tuple val(meta), path(bam)
+
+    output:
+    tuple val(meta), path("*.bam"), emit: bam
+    path "versions.yml"          , emit: versions
+
+    script:
+    def prefix = options.suffix ? "${meta.id}${options.suffix}" : "${meta.id}"
+
```

Comment:
I would be a bit concerned that due to the standard nf-core/modules, the BAM input (`$bam`) would often have the same `${prefix}.bam` naming system as the output file used in this process. It would be good to include the warning in  the last bullet of the [`Naming conventions`](https://nf-co.re/developers/adding_modules#naming-conventions)

Reply:
I have added that check now (couldn't figure out how to do it in response to this comment though).